src/utils.jl

```diff
@@ -185,12 +185,31 @@ function __get_bcresid_prototype(::StandardBVProblem, prob::BVProblem, u)
     return prototype, size(prototype)
 end
 
-@inline function __fill_like(v, x, args...)
+@inline function __similar(x, args...)
     y = similar(x, args...)
+    return __init_bigfloat_array!!(y)
+end
+
+@inline function __init_bigfloat_array!!(x)
+    if ArrayInterface.can_setindex(x)
+        eltype(x) <: BigFloat && fill!(x, BigFloat(0))
+        return x
+    end
+    return x
+end
+
+@inline function __fill_like(v, x)
+    y = __similar(x)
     fill!(y, v)
     return y
 end
-@inline __zeros_like(args...) = __fill_like(0, args...)
+@inline function __zeros_like(u)
+    if ArrayInterface.can_setindex(u)
+        eltype(u) <: BigFloat && __fill_like(BigFloat(0), u)
+        return u
```

Comment:
missing a `u .= 0`?

Reply:
There is a `return __fill_like(0, u)` below

Reply:
There is a `return __fill_like(0, u)` below